<?xml version="1.0"?>

<launch>
    <!-- arguments -->
    <arg name="use_rviz" default="true" />
    <!--PLAY BAG FILE-->
    <param name="use_sim_time" value="true" />
    <node pkg="rosbag" type="play" name="player" args="$(find lynmax)/bags/2017-11-07-17-16-07.bag --clock"/>
    <rosparam command="load" file="$(find lynmax)/launch/param/outdoor/GPSIMU_ekf_params.yaml" />
    <rosparam command="load" file="$(find lynmax)/launch/param/outdoor/navsat_params.yaml" />

    <node name="desktop"   pkg="lynmax"  type="Desktop.py" output="screen"  />

    <!-- Map server -->
    <node name="map_server" pkg="map_server" type="map_server" args="$(find lynmax)/map/test1108/map.yaml">
    </node>

    <!-- TF setting -->
    <include file="$(find lynmax)/launch/includes/racecar_tf.launch.xml" />

    <!-- ODOMETRY -->
    <!--rf2o_Laser_Odometry-->
    <include file="$(find lynmax)/launch/includes/rf2o.launch.xml" />
    <!-- Robot_Localization -->
    <node pkg="robot_localization" type="ekf_localization_node" name="ekf_se_odom" clear_params="true">
	<remap from="/base_odom" to="/Desktop/base_odom"/>
	<remap from="/imu" to="/Desktop/imu"/>
    </node>

    <node pkg="robot_localization" type="ekf_localization_node" name="ekf_se_map" clear_params="true">
	<remap from="odometry/filtered" to="odometry/filtered_map"/>
	<remap from="/base_odom" to="/Desktop/base_odom"/>
	<remap from="/imu" to="/Desktop/imu"/>
    </node>

    <node pkg="robot_localization" type="navsat_transform_node" name="navsat_transform_node" clear_params="true" respawn="true" output="screen" >
        <rosparam param="datum">[24.822585, 121.767084, 0.0, map, base_footprint]</rosparam>
        <remap from="odometry/filtered" to="odometry/filtered_map"/>
        <remap from="gps/fix" to="/Desktop/ublox_gps/fix"/>
	<remap from="imu/data" to="/Desktop/imu"/>
    </node>

    <!-- Navstack -->
    <node pkg="move_base" type="move_base" respawn="true" name="move_base" output="screen" > <!--output="screen"-->
    
      <rosparam file="$(find lynmax)/launch/param/outdoor/costmap_common_params.yaml" command="load" ns="global_costmap" />
      <rosparam file="$(find lynmax)/launch/param/outdoor/costmap_common_params.yaml" command="load" ns="local_costmap" />
      <rosparam file="$(find lynmax)/launch/param/outdoor/local_costmap_params.yaml" command="load" />
      <rosparam file="$(find lynmax)/launch/param/outdoor/global_costmap_params.yaml" command="load" />
      <rosparam file="$(find lynmax)/launch/param/outdoor/base_local_planner_params.yaml" command="load" />
  
      <rosparam file="$(find lynmax)/launch/param/outdoor/move_base_params.yaml" command="load" />
      
      <param name="base_global_planner" type="string" value="navfn/NavfnROS" />
      <param name="base_local_planner" value="base_local_planner/TrajectoryPlannerROS"/>

      <remap from="odom" to="/odometry/filtered_map" />

    </node>   

    <!-- Rviz -->
    <node pkg="rviz" type="rviz" name="rviz" args="-d $(find lynmax)/rviz/lynmax_nav_satellite.rviz" if="$(arg use_rviz)" />
</launch>
